### PR TITLE
Dockerfile for building images for Docker and Kubernetes.

### PR DESCRIPTION
**Pull request by tobiasake**
### Description of the Change
Adding a Dockerfile so user can build Docker image for Docker and Kubernetes. 

### Benefits
Make it easy to build RemRem-Publish docker images.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by:

### CODE REVIEW

File: publish-service/src/main/docker/README.txt [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+docker build -t remrem-publish:0.6.8 --build-arg URL=https://jitpack.io/com/github/Ericsson/eiffel-remrem-publish/publish-service/0.6.8/publish-service-0.6.8.war -f src/main/docker/Dockerfile .
+
+
+
+B: Build RemRem-Publish based on local RemRem-Publish source code changes
+1. Build RemRem-Publish service artifact:
+cd (git root dir)
+mvn package -DskipTests
+
+2. Build RemRem-Publish Docker image:
```
Comment (e-pettersson-ericsson) [inline publish-service/src/main/docker/README.txt]: The header here can be more precise; and mention the locally built image (based on latest source code changes).

File: wiki/markdown/docker.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+2. Change to publish-service directory: 
+`cd publish-service`
+
+3. Build the Docker image with the war file that was produced from previous step: 
+
+
+`docker build -t remrem-publish:0.6.9 --build-arg URL=./target/publish-service-0.6.9.war -f src/main/docker/Dockerfile .` 
+
+
+Now docker image has build with tag "remrem-publish:0.6.9"
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/docker.md]: "is built", or "has been built"

File: wiki/markdown/docker.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+## Run Docker image on local Docker Host
+To run the produced docker image on the local Docker host, execute this command: 
+
+
+`docker run -p 8080:8080 --expose 8080 -e server.port=8080 -e logging.level.log.level.root=DEBUG -e logging.level.org.springframework.web=DEBUG -e logging.level.com.ericsson.ei=DEBUG remrem-publish:0.6.9`
+
+RabbitMq and other RemRem-Publish required components need to running and configured via these application properties that is provided to the docker command above. See the application.properties file for all available/required properties:
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/docker.md]: need to be running

File: wiki/markdown/docker.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+[application.properties](https://github.com/eiffel-community/eiffel-remrem-publish/blob/master/publish-service/src/main/resources/application.properties)
+
+# Some info of all flags to this command
+
+
+## RemRem-Publish Spring Properties
+
+
+<B>"-e server.port=8080"</B> - Is the Spring property setting for RemRem-Publish applications web port.
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/docker.md]: application in singular

File: wiki/markdown/docker.md (hunk 3 above)
Comment (e-pettersson-ericsson) [inline wiki/markdown/docker.md]: what does "**these** application properties" refer to? Remove the "these" and I think it becomes more clear to read.

File: publish-service/src/main/docker/README.txt [hunk 5]
```diff
@@ -0,0 +1,16 @@
+A: Build RemRem-Publish Docker image based on RemRem-Publish from an Artifactory, e.g. Jitpack:
+cd (git root dir)/publish-service
+docker build -t remrem-publish:0.6.8 --build-arg URL=https://jitpack.io/com/github/Ericsson/eiffel-remrem-publish/publish-service/0.6.8/publish-service-0.6.8.war -f src/main/docker/Dockerfile .
```
Comment (e-pettersson-ericsson) [inline publish-service/src/main/docker/README.txt]: Same comment here, as in remrem-generate about the url.

File: wiki/markdown/docker.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+`docker build -t remrem-publish:0.6.9 --build-arg URL=./target/publish-service-0.6.9.war -f src/main/docker/Dockerfile .` 
+
+
+Now docker image has build with tag "remrem-publish:0.6.9"
+
+## Run Docker image on local Docker Host
+To run the produced docker image on the local Docker host, execute this command: 
+
+
+`docker run -p 8080:8080 --expose 8080 -e server.port=8080 -e logging.level.log.level.root=DEBUG -e logging.level.org.springframework.web=DEBUG -e logging.level.com.ericsson.ei=DEBUG remrem-publish:0.6.9`
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/docker.md]: See my comment about ports in the PR for remrem-generate

File: wiki/markdown/docker.md (hunk 2 above)
Comment (tobiasake) [reply wiki/markdown/docker.md]: Fixed.

File: wiki/markdown/docker.md (hunk 3 above)
Comment (tobiasake) [reply wiki/markdown/docker.md]: fixed.

File: wiki/markdown/docker.md (hunk 4 above)
Comment (tobiasake) [reply wiki/markdown/docker.md]: Fixed.

File: wiki/markdown/docker.md (hunk 3 above)
Comment (tobiasake) [reply wiki/markdown/docker.md]: Fixed.

File: publish-service/src/main/docker/README.txt (hunk 5 above)
Comment (tobiasake) [reply publish-service/src/main/docker/README.txt]: fixed.

File: wiki/markdown/docker.md (hunk 6 above)
Comment (tobiasake) [reply wiki/markdown/docker.md]: RemRem-Generate documentation will be changed to port number 8081 or something.

File: publish-service/src/main/docker/health-check.sh [hunk 7]
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+#NC_BIN=nc.traditional
+#NC_BIN=nc
+SLEEP_LENGTH=8
+
+WAIT_HOSTS="${@}"
+
+echo "Will wait for fallowing hosts getting started:"
```
Comment (Christoffer-Cortes) [inline publish-service/src/main/docker/health-check.sh]: following

File: publish-service/src/main/docker/start-service.sh [hunk 8]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+# These two varibales need to be set from Docker-compose or K8S at startup if MB and/or DB healthcheck should be used.
```
Comment (Christoffer-Cortes) [inline publish-service/src/main/docker/start-service.sh]: variables

File: publish-service/src/main/docker/health-check.sh (hunk 7 above)
Comment (tobiasake) [reply publish-service/src/main/docker/health-check.sh]: fixed.

File: publish-service/src/main/docker/start-service.sh (hunk 8 above)
Comment (tobiasake) [reply publish-service/src/main/docker/start-service.sh]: fixed.

**Pull request merged**